### PR TITLE
Differences between a Hub, Switch, and Router

### LINKED ISSUE #2358: Difference Between a Hub, Switch and Router
We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner. Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published.

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

### Proposed title of article:
Difference Between a Hub, Switch and Router

### Introduction paragraph (2-3 paragraphs):
Hubs, switches, and routers are all common types of devices to help connect a network. Because they are often integrated into a single device, it can be challenging to understand the distinction between these items. In some cases, the terms may even be used interchangeably, though this is incorrect. The functions and uses of these devices are all very different from one another. 

In this article, we will look at a simple guide on the difference between a hub, switch, and router to others understand their dissimilarity.

### Key takeaways: 
1. What is a hub, switch and  router.
2. How each transmit data from end to another
3. Connectivity - What does each of these items connect
4. The security of each item.
5. Networks used - The uses of each item.
### References: 
N/A

### PR DESCRIPTION
**Pull request by Jeddy947**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
This fixes #2358

### CODE REVIEW

**Comment by hectorkambow** [comment]
@Jeddy947 
Good afternoon and thank you for submitting your article for publishing(and welcome to the EngEd program). Im Hector - one of the content moderators that may be doing your Final review on the article before it is published.

Your article has been entered into our queue and should be reviewed (and published pending any changes) as soon as a peer reviewer is finished with the ones in the queue before it. 🚀

Thank you.

**Comment by Bayler** [comment]
@Jeddy947 Welcome to the Section Engineering community. I will be your reviewer. Please note that your 1st article is going to serve as a guide to improve (if needed) any future PRs. 

Regards,
Willies

File: content/articles/switch-vs-router-vs-hub/index.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
+In networking the terms "Switches", "Hubs", and "Routers" are sometimes used interchangeably and this is wrong. Despite them being similar, there are differences in how they handle data. These three components may be integrated into a single device making it hard for a student to distinguish between them.
+
+In this article, we are going to discuss each device and their functionalities in a network.
+
+### Switch
```
Comment (Bayler) [inline content/articles/switch-vs-router-vs-hub/index.md:5]: Please add a "Table of contents" section for the reader in between the sentence and the header. It is important to always include one inside your article. Also make sure they link to their respective headers

File: content/articles/switch-vs-router-vs-hub/index.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
+In networking the terms "Switches", "Hubs", and "Routers" are sometimes used interchangeably and this is wrong. Despite them being similar, there are differences in how they handle data. These three components may be integrated into a single device making it hard for a student to distinguish between them.
+
+In this article, we are going to discuss each device and their functionalities in a network.
+
+### Switch
+A switch is a multicast networking device that works under the Datalink layer of the OSI model and connects a bunch of computers or devices in a network.
+
```
Comment (Bayler) [inline content/articles/switch-vs-router-vs-hub/index.md:7]: The definition of a switch is too summarized. Please include some more information

File: content/articles/switch-vs-router-vs-hub/index.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+### Disadvantages of using hubs
+- It increases the chances of collision domains between packets when being transferred from one device to another.
+- Hubs operate under half-duplex that's only one occasion of data transmission at a time.
+- Hubs share data to all the devices in a network thus insecure.
+- Hubs does lots of wastage of bandwidth when transmitting data.
+
+### Switch Vs Hub
+Hub is a broadcast device that sends data from one node to all but Switch is a multicast device that can send data to a particular node you want.
+
```
Comment (Bayler) [inline content/articles/switch-vs-router-vs-hub/index.md:38]: I believe there are more comparisons between a switch and a hub. Please add them.

File: content/articles/switch-vs-router-vs-hub/index.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+#### Souter
+It is commonly used in Local Area Network and Metropolitan Area Network(MAN).
+
+#### hub
+It is similar to Switch because it is used in the local area network(LAN).
+
+### Mode of data transmission
+It defines the direction in which data flow between two communicating devices. There are three types of transmission modes. Read this [documentation](https://afteracademy.com/blog/what-are-the-data-transmission-modes-in-a-network) for more information on the Mode of data transmission
+
```
Comment (Bayler) [inline content/articles/switch-vs-router-vs-hub/index.md:88]: In this section, you need to first introduce the three modes of transmission then direct the reader to the documentation. It is good practice to do so.

File: content/articles/switch-vs-router-vs-hub/index.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+### Application of each Device
+#### Switch
+It is commonly used in local area networks for connecting many nodes.
+
+#### Souter
+It is commonly used in Local Area Network and Metropolitan Area Network(MAN).
+
+#### hub
+It is similar to Switch because it is used in the local area network(LAN).
+
```
Comment (Bayler) [inline content/articles/switch-vs-router-vs-hub/index.md:85]: This section is too summarized. Switches, hubs, and routers have more applications and functions besides the listed ones. Make sure you've exhausted the points. This is beneficial to the reader.

File: content/articles/switch-vs-router-vs-hub/index.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+### Transmission of Data
+#### Switch
+Switch transmit data from one device to another in form of [frames](https://en.wikipedia.org/wiki/Frame_(networking)).
+
+#### Router 
+The router transmits data from one network to another in form of [packets](https://www.cloudflare.com/learning/network-layer/what-is-a-packet/).
+
+#### Hub
+Hub transmits data from one device to another in form of binary bits.
+
```
Comment (Bayler) [inline content/articles/switch-vs-router-vs-hub/index.md:117]: Finally, you could add a summary or conclusion section to summarize the points in your article. It's always good practice to add that section.

**Comment by Bayler** [comment]
Hey @Jeddy947 
Welcome once again to Section - EngEd. 
I have enjoyed reading the article. It's very informative. 
Good job! Keep it up! 
I have requested a few initial changes to be made so that we can proceed with the grammar review. Please do so and ping me here once you are done.

Once again good job.

**Comment by Bayler** [comment]
@Jeddy947 Please work on the revisions.

**Comment by Jeddy947** [comment]
@Bayler Sorry for the delay.  Am working on the changes.

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by Jeddy947** [comment]
@Bayler I have done the changes. Have a look please

**Comment by Bayler** [comment]
Hi @Jeddy947 I'd like to welcome you again to our Engineering community. Most first article usually have a lot of revisions, but  that should never worry you, you'll get the hang of it with time. I want you to take note of the commits I've made in your article so that you don't repeat the same mistakes in your next article; especially on how to link the table of content to their respective headers. 

Otherwise, great job! See you on the next one.

Regards,
@Bayler

**Comment by LinusMuema** [comment]
Hi @Jeddy947 
I see @Bayler has given you several welcoming messages and I couldn't have done it better. We are glad to have you in this program. 
Just check my edits for future articles and you will hopefully grow to be good at this.

cc: @Bayler

**Pull request merged**